v3/src/components/graph/adornments/lsrl/lsrl-adornment-model.ts

```diff
@@ -131,20 +140,19 @@ export const LSRLAdornmentModel = AdornmentModel
   }
 }))
 .actions(self => ({
-  updateLines(
-    line: {category?: string, intercept?: number, rSquared?: number, slope?: number, sdResiduals?: number},
-    key="", index?: number
-  ) {
+  updateLines(line: ILSRLLine, key="", index?: number) {
+    const { category, equationCoords, intercept, rSquared, sdResiduals, slope } = line
     const existingLines = self.lines.get(key)
     const newLines = existingLines ? [...existingLines] : []
     // Remove any pre-existing line in newLines at specified index, otherwise we can end up with duplicates.
-    index && newLines.splice(index, 1)
+    isFiniteNumber(index) && newLines.splice(index, 1)
```

Comment:
This line wasn't working when `index` was `0`. Fixing that meant the former line 168 `self.lines.clear()` was no longer needed. I believe I must've originally added that because this line (148) wasn't doing what I thought it was but was "failing" silently.

Reply:
Since it's not really an issue of floating-point finiteness, `;(index != null) && newLines.splice(index, 1)` might be clearer.